src/components/utils/OffchainSigner/SignerKeyringManager.ts

```diff
@@ -0,0 +1,81 @@
+import { Keyring } from '@polkadot/api'
+import { hexToU8a, stringToU8a, u8aToHex, u8aToString } from '@polkadot/util'
+import { keccakAsU8a, naclDecrypt, naclEncrypt } from '@polkadot/util-crypto'
+
+class SignerKeyringManager {
+  private readonly keyring: Keyring
+
+  constructor() {
+    this.keyring = new Keyring({ type: 'sr25519' })
+  }
+
+  private generateSalt() {
+    // has to be 32 bytes, otherwise naclEncrypt will throw an error
+    const SALT_LENGTH = 32
+    const arr = new Uint8Array(SALT_LENGTH)
+    const arrayResult = crypto.getRandomValues(arr)
+
+    return arrayResult
+  }
+
+  private generateSecret(password: string, inputSaltStr?: string) {
+    const salt = inputSaltStr ? hexToU8a(inputSaltStr) : this.generateSalt()
+    const saltStr = u8aToHex(salt)
+    const secret = keccakAsU8a(saltStr + password)
+
+    return {
+      saltStr,
+      secret,
+    }
+  }
+
+  private generateNonce() {
+    const HEX_CONSTANT = '0x80001f'
```

Comment:
extract all const to top of the file

Reply:
Extracted in [e09121f](https://github.com/dappforce/polkaverse/pull/74/commits/e09121f767f49e57bde153dea1342819b746732d)